src/dllinject/dllinject.c

```diff
@@ -1474,22 +1477,44 @@ static int canon_path(const wchar_t *file, int filelen, char *dest)
 		goto out_empty;
 	}
 
-	wcscpy(widepath, backslash_prefix);
-
+	/* Two wchar_t buffers with a size of WIDE_PATH_MAX take up 128 KiB in
+	 * total, which might exceed the stack reserve size defined by the target
+	 * executable. Therefore, determine how much we actually need before
+	 * alloca()ing that exact amount of memory. */
 	if(wcsncmp(file, other_prefix, prefix_len) == 0 ||
 	   wcsncmp(file, backslash_prefix, prefix_len) == 0) {
+		branch = 1;
+		widepath_len = filelen;
+	} else if(is_full_path(file)) {
+		branch = 2;
+		widepath_len = prefix_len + filelen;
+	} else {
+		branch = 3;
+		widepath_len = prefix_len + GetCurrentDirectoryW(0, NULL) + 1 + filelen;
+	}
+
+	bufsize = (widepath_len + 1) * sizeof(wchar_t);
+	widepath = __builtin_alloca(bufsize);
+	if(widepath) {
```

Comment:
This should be if(!widepath), right? This patch doesn't work for me unless I change this. Are you sure you posted the patch that you were testing with?